src/nvim/charset.c

```diff
@@ -82,7 +82,7 @@ int buf_init_chartab(buf_T *buf, int global)
 {
   int c;
   int c2;
-  char_u *p;
+  unsigned char *p;
```

Comment:
I suppose that eventually: all public functions in charset will take `char *` arguments. These will be converted to `uint_8 *`internally in functions doing unsigned operations. But this will be a larger refactor that requires changing all call sites. Leaving it `char_u *` is fine for this PR.